pkg/installation/utils.go

```diff
@@ -0,0 +1,232 @@
+	"strings"
+	"time"
+
+	"github.com/Masterminds/semver"
+	docker "github.com/fsouza/go-dockerclient"
+	"github.com/kyma-project/cli/internal/minikube"
+	"github.com/mitchellh/mapstructure"
+	"github.com/pkg/errors"
+	git "gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/storage/memory"
+	v1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func (i *Installation) applyResourceFile(filepath string) error {
+	_, err := i.getKubectl().RunCmd("apply", "-f", filepath)
+	return err
+}
+
+func (i *Installation) buildKymaInstaller(imageName string) error {
+	dc, err := minikube.DockerClient(i.Options.Verbose, i.Options.LocalCluster.Profile)
+	if err != nil {
+		return err
+	}
+
+	var args []docker.BuildArg
+	return dc.BuildImage(docker.BuildImageOptions{
+		Name:         strings.TrimSpace(string(imageName)),
+		Dockerfile:   filepath.Join("tools", "kyma-installer", "kyma.Dockerfile"),
+		OutputStream: ioutil.Discard,
+		ContextDir:   filepath.Join(i.Options.LocalSrcPath),
+		BuildArgs:    args,
+	})
+}
+
+func (i *Installation) checkIfResourcePresent(namespace, kind, name string) error {
+	_, err := i.getKubectl().RunCmd("-n", namespace, "get", kind, name)
+	return err
+}
+
+func (i *Installation) getInstallationStatus() (status string, desc string, err error) {
+	status, err = i.getKubectl().RunCmd("get", "installation/kyma-installation", "-o", "jsonpath='{.status.state}'")
+	if err != nil {
+		return
+	}
+	desc, err = i.getKubectl().RunCmd("get", "installation/kyma-installation", "-o", "jsonpath='{.status.description}'")
+	return
+}
+
+func (i *Installation) printInstallationErrorLog() error {
+	logs, err := i.getKubectl().RunCmd("get", "installation", "kyma-installation", "-o", "go-template", "--template={{- range .status.errorLog -}}{{printf \"%s:\n %s [%s]\n\" .component .log .occurrences}}{{- end}}")
+	if err != nil {
+		return err
+	}
+	fmt.Println(logs)
+	return nil
+}
+
+func (i *Installation) getMasterHash() (string, error) {
+	ctx, timeoutF := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer timeoutF()
+	r, err := git.CloneContext(ctx, memory.NewStorage(), nil,
+		&git.CloneOptions{
+			Depth: 1,
+			URL:   "https://github.com/kyma-project/kyma",
+		})
+	if err != nil {
+		return "", err
+	}
+	h, err := r.Head()
+	if err != nil {
+		return "", err
+	}
+
+	return h.Hash().String()[:8], nil
+}
+
+func (i *Installation) setAdminPassword() error {
+	if i.Options.Password == "" {
+		return nil
+	}
+	encPass := base64.StdEncoding.EncodeToString([]byte(i.Options.Password))
+	_, err := i.k8s.Static().CoreV1().ConfigMaps("kyma-installer").Patch("installation-config-overrides", types.JSONPatchType,
+		[]byte(fmt.Sprintf("[{\"op\": \"replace\", \"path\": \"/data/global.adminPassword\", \"value\": \"%s\"}]", encPass)))
+	if err != nil {
+		err = errors.Wrap(err, "Error setting admin password")
+	}
+	return err
+}
+
+func removeActionLabel(acc *[]map[string]interface{}) error {
+	for _, config := range *acc {
```

Comment:
This function can also be smaller by not using continue statements.

Reply:
Done